src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -59,9 +59,13 @@ def get_remaining_time_in_minutes(self, current_time=None, log_to_stdout=False):
 
         return remaining_time_in_minutes
 
-    def is_package_install_time_available(self, remaining_time_in_minutes=None):
+    def is_package_install_time_available(self, remaining_time_in_minutes=None, number_of_packages=1):
         """Check if time still available for package installation"""
-        cutoff_time_in_minutes = Constants.REBOOT_BUFFER_IN_MINUTES + Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES
+        cutoff_time_in_minutes = Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES * number_of_packages
+
+        if Constants.REBOOT_SETTINGS[self.execution_config.reboot_setting] != Constants.REBOOT_NEVER:
```

Comment:
can self.execution_config.reboot_setting be None? @rane-rajasi @kjohn-msft 
Should I add a check if self.execution_config.reboot_setting is None before accessing it?
If it can be None, should we add reboot time in cutoff?
I think it should not be None as per file core_logic\ExecutionConfig.py, line no. 64. But wanted to confirm it.

Reply:
No self.execution_config.reboot_setting will not be None at this point in the execution flow. Line 64 in ExecutionConfig.py (init) will throw an exception in case it is None and end the execution